wandb/integration/keras_core/model_checkpoint.py

```diff
@@ -0,0 +1,129 @@
+import os
+import sys
+from typing import Dict, List, Optional, Union
+
+import wandb
+from wandb.sdk.lib import telemetry
+from wandb.sdk.lib.paths import StrPath
+
+try:
+    from keras_core.callbacks import ModelCheckpoint
+except Exception as e:
+    wandb.Error(e)
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+Mode = Literal["auto", "min", "max"]
+SaveStrategy = Literal["epoch"]
+
+
+def _log_artifact(
+    filepath: StrPath, aliases: Optional[List] = None, metadata: Optional[Dict] = None
+):
+    aliases = ["latest"] if aliases is None else aliases + ["latest"]
+    metadata = wandb.run.config.as_dict() if metadata is None else metadata
+    model_checkpoint_artifact = wandb.Artifact(
+        f"run_{wandb.run.id}_model", type="model", metadata=metadata
+    )
+    if os.path.isfile(filepath):
+        model_checkpoint_artifact.add_file(filepath)
+    elif os.path.isdir(filepath):
+        model_checkpoint_artifact.add_dir(filepath)
+    else:
+        raise FileNotFoundError(f"No such file or directory {filepath}")
+    wandb.log_artifact(model_checkpoint_artifact, aliases=aliases or [])
+
+
+class WandbModelCheckpoint(ModelCheckpoint):
+    """`WandbModelCheckpoint` automatically logs model checkpoints to W&B and versions
+    them as [W&B artifacts](https://docs.wandb.ai/guides/artifacts).
+
+    Since this callback is subclassed from
+    [`keras_core.callbacks.ModelCheckpoint`](https://keras.io/keras_core/api/callbacks/model_checkpoint/),
+    the checkpointing logic is taken care of by the parent callback.
+
+    This callback is to be used in conjunction with training using `model.fit()` to save
```

Comment:
This callback is to be used in conjunction with training using `model.fit()`

correction:

This callback is to be used in conjunction with `model.fit()`